zeekclient/types.py

```diff
@@ -582,11 +582,14 @@ def from_config_parser(cls, cfp, _section=None):
                         config.instances.append(Instance(key))
                     else:
                         hostport = val
-                        parts = hostport.split(':', 1)
-                        if len(parts) != 2 or not parts[0] or not parts[1]:
+                        parts = hostport.rpartition(':')
+                        if(parts[0] == '' or parts[1] == ''):
```

Comment:
Did you intend to check `parts[2]` instead of `parts[1]` here?

I'd suggest you directly bind the result of `rpartition` above to variables to make the code easier to follow and harder to mess up, e.g.,

```python
host, _, port = hostport.rpartition(':')
```

You can then directly use and modify these variables below.

Reply:
Oh, 100%.  I also hadn't updated the variables in the log error.